File: server/src/graphs/state.js
// server/src/graphs/state.js
import { Annotation, MessagesAnnotation } from '@langchain/langgraph';

export const GraphState = Annotation.Root({
  ...MessagesAnnotation.spec,

  userInput: Annotation({
    reducer: (_, next) => next,
    default: () => '',
  }),

  // order | knowledge | general
  intent: Annotation({
    reducer: (_, next) => next,
    default: () => '',
  }),

  orderResult: Annotation({
    reducer: (_, next) => next,
    default: () => null,
  }),

  ragResult: Annotation({
    reducer: (_, next) => next,
    default: () => '',
  }),

  finalAnswer: Annotation({
    reducer: (_, next) => next,
    default: () => '',
  }),
});

File: server-py/app/graphs/state.py
"""LangGraph 状态定义"""
from typing import Annotated, Optional

from langgraph.graph.message import add_messages
from typing_extensions import TypedDict


class GraphState(TypedDict):
    messages: Annotated[list, add_messages]

    user_input: str

    # order | knowledge | general
    intent: str

    order_result: Optional[dict]
    rag_result: str
    final_answer: str
